core/lib/dal/src/blocks_dal.rs

```diff
@@ -2148,6 +2148,21 @@ impl BlocksDal<'_, '_> {
         .await?
         .map(|row| row.virtual_blocks as u32))
     }
+
+    pub async fn get_batch_pubdata(
```

Comment:
Bikeshedding: I feel that DAL should not be responsible for domain logic (mapping L1 batch metadata to pubdata in this case). It would be better to use `get_l1_batch_metadata()` in the endpoint handler directly.

Reply:
LGTM, I'll file an issue in our fork to tackle this. Thanks!

Reply:
We created this issue to tackle this https://github.com/lambdaclass/zksync-era/issues/146